### PR TITLE
Core/Weather: Port weather changes from master

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed:**

-  Port 6f7d048 and 6eb9973 from master to 3.3.5
-  convert SMSG_WEATHER to new system
-  Fixed weather updates to be thread safe

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.) Build (on Windows), tested

### CODE REVIEW

**Comment by jackpoz** [comment]
what does this PR fix in 335 ? are there any weather issues ?

**Comment by Shauren** [comment]
Thread safety - WeatherMgr::AddWeather in Player::Update

File: src/server/game/Server/WorldSession.h [hunk 1]
```diff
@@ -81,6 +81,10 @@ namespace WorldPackets
         class Hello;
         class TrainerBuySpell;
     }
+    namespace Misc
+    {
+        class Weather;
```
Comment (Shauren) [inline src/server/game/Server/WorldSession.h]: still not a proper cherry-pick - this line does not exist in master and forward declaring in WorldSession is not needed for SMSG packets

Comment (Shauren) [reply src/server/game/Server/WorldSession.h]: I'll just do it myself later today

**Comment by Shauren** [comment]
Side note - the includes you have in MiscPackets.h are from the pre-include-cleanup times, this is what you get when doing cherry picks completely out of order

**Comment by ForesterDev** [comment]
@Shauren, when I cherry-picked commit 6f7d048765f102110cbf9ad4653c078fbadaa105, I get the whole files MiscPackets with other unnecessary packet. I've cleaned they by hands, but headers left untouched

**Pull request merged**